src/histogram/histogram.cpp

```diff
@@ -49,8 +49,9 @@ py::class_<bh::histogram<A, S>> register_histogram_by_type(py::module& m, const
     .def(py::self + py::self)
     .def(py::self == py::self)
     .def(py::self != py::self)
-    .def(py::self *= double())
-    .def(py::self /= double())
+    // fail on atomics
+    // .def(py::self *= double())
+    // .def(py::self /= double())
```

Comment:
At some point, code should be added which statistically checks what operations the accumulator/the storage supports and then enable those. This can be done based on boost::mp11::mp_valid.

See
https://github.com/boostorg/histogram/blob/7301d6a4c2d2500245648388ab7a9709985a1f4f/include/boost/histogram/detail/meta.hpp#L137
and the following lines 215, 218, 221, 224, 227.

You would use these here like so:
```
if (has_operator_rmul<histogram_t, double>::value)
   hist.def(py::self *= double());
```

Reply:
It may be worthwhile to duplicate the code in your library, so that you don't rely on boost::histogram's details.

Reply:
I am locked to specific checkouts of Boost::Histogram, so in general this is pretty safe for development for now. Once we hit beta stage or so, I could internalize details, or maybe we could mark this with a comment mentioning that it's used directly by the Python bindings as well, so if someone changes it they know to make the required changes in boost-histogram. ;)

Reply:
I think it is ok to duplicate the code, to disentangle the two works. Ideally, boost-histogram should only depend on the public interface of boost::histogram. Right now we cannot have that, but for the future it would be nice.